sql/core/src/main/scala/org/apache/spark/sql/catalog/Catalog.scala

```diff
@@ -102,6 +102,83 @@ abstract class Catalog {
   def listColumns(dbName: String, tableName: String): Dataset[Column]
 
   /**
+   * Find the database with the specified name. This returns [[None]] when no [[Database]] can be
+   * found.
+   *
+   * @since 2.1.0
+   */
+  def findDatabase(dbName: String): Option[Database]
```

Comment:
Per discussion offline, it'd be better to change this to throw an analysis exception for java compatibility, to be consistent with rest of Spark.

We should annotate the method with a throw annotation.